gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/oauth2/OAuth2SecurityController.groovy

```diff
@@ -69,25 +72,34 @@ class OAuth2SecurityController {
       throw new AccessDeniedException("OAuth2 access denied")
     }
 
+    // get token
     def form = new LinkedMultiValueMap<String, String>();
     form.add('code', code)
     form.add('client_id', oAuth2ProtectedResourceDetails.clientId)
     form.add('client_secret', oAuth2ProtectedResourceDetails.clientSecret)
     form.add('redirect_uri', request.scheme + '://' + request.serverName + ':' + request.serverPort + '/oauth/code')
     form.add('grant_type', 'authorization_code')
-    ResponseEntity<DefaultOAuth2AccessToken> responseEntity = restTemplate.postForEntity(oAuth2ProtectedResourceDetails.accessTokenUri, form, DefaultOAuth2AccessToken)
-
-    PreAuthenticatedAuthenticationToken authn = new PreAuthenticatedAuthenticationToken(new User(email: 'jacobkiefer'), responseEntity.body)
+    ResponseEntity<DefaultOAuth2AccessToken> tokenResponse = restTemplate.postForEntity(oAuth2ProtectedResourceDetails.accessTokenUri, form, DefaultOAuth2AccessToken)
+
+    // get email
+    HttpHeaders headers = new HttpHeaders()
+    headers.set('Authorization', 'Bearer ' + tokenResponse.body.value)
+    def infoReq = new HttpEntity(headers)
+    ResponseEntity<Map> userInfo = restTemplate.exchange('https://www.googleapis.com/oauth2/v2/userinfo', HttpMethod.GET, infoReq, Map)
```

Comment:
Don't hardcode - pull into param. See various places that use the @Value annotation